onnx/shape_inference/shape_inference.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include "onnx/proto_utils.h"
+#include "onnx/defs/schema.h"
+
+namespace ONNX_NAMESPACE { namespace shape_inference {
+
+struct InferenceContextImpl : public InferenceContext {
+  virtual const AttributeProto* getAttribute(const std::string& name) const {
+    auto iter = attributesByName_.find(name);
+    if (iter == attributesByName_.end()) {
+      return nullptr;
+    } else {
+      return iter->second;
+    }
+  }
+  virtual size_t getNumInputTypes() const {
+    return allInputTypes_.size();
+  }
+  virtual const TypeProto_Tensor* getInputType(size_t index) const {
+    return allInputTypes_[index];
+  }
+  virtual size_t getNumOutputTypes() const {
+    return allOutputTypes_.size();
+  }
+  virtual TypeProto_Tensor* getOutputType(size_t index) {
+    return &allOutputTypes_[index];
+  }
+  std::unordered_map<std::string, const AttributeProto *> attributesByName_;
+  std::vector<const TypeProto_Tensor*> allInputTypes_;
+  std::vector<TypeProto_Tensor> allOutputTypes_;
+};
+
+void InferShapes(ONNX_NAMESPACE::ModelProto& m) {
```

Comment:
nit, we don't need `ONNX_NAMESPACE` here?

Reply:
++